lib/srv/reexec.go

```diff
@@ -1106,6 +1107,29 @@ func ConfigureCommand(ctx *ServerContext, extraFiles ...*os.File) (*exec.Cmd, er
 		cmd.ExtraFiles = append(cmd.ExtraFiles, extraFiles...)
 	}
 
+	// For remote port forwarding, the child needs to run as the user to
+	// create listeners with the correct permissions.
+	if subCommand == teleport.RemoteForwardSubCommand {
+		localUser, err := user.Lookup(ctx.Identity.Login)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		// Ensure that the working directory is one that the child has access to.
+		cmd.Dir = localUser.HomeDir
```

Comment:
Why is this necessary?

Could this have unintended consequences? What if the user doesn't have a home?

Reply:
Default behavior is for the child process to inherit the workdir from the parent, but if the processes are running under different users (i.e. root parent and non-root child) the child will get a permission error and port forwarding will fail.

If the user doesn't have a home, they should be dropped at root, but I updated this to drop everyone at root so we don't have to worry about it.